resources/js/home.js

```diff
@@ -1,73 +1,93 @@
 
-var fromMenuObj
-var indexContentObj
-var paraHeight
+let paraHeight
+let headerObj, contentObj
 
 
-// Loading Contents for Homepage
-const contentRequest = new XMLHttpRequest()
+/*--------------- Home page content loading section ---------------*/
 const para = document.querySelector('#blog-paragraph')
 const imageArea = document.querySelector('.image-area')
 
-contentRequest.onload = function () {
-  if (this.status === 200) {
-    try {
-      indexContentObj = JSON.parse(this.responseText)
-      para.textContent = indexContentObj.Content
-
-      const image = document.createElement('img')
-      image.src = indexContentObj.image
-      image.className += 'content-img';
-      imageArea.appendChild(image)
-
-      paraResize()
-      
-    } catch {
-      console.warn('JSON not parsed')
-    }
-  } else {
-    console.warn('File not found')
-  }
-}
+utils.jsonCaller('get', 'resources/json/homepage.json', function (object) {
+  para.textContent = object.content
+  // let headerObj = object.table
+  headerObj = object.table
+
+  const image = document.createElement('img')
+  image.src = object.image
+  image.className += 'content-img'
+  imageArea.appendChild(image)
+
+  paraResize()
+  loadTable(headerObj)
+})
 
-contentRequest.open('get', 'resources/json/homepage.json')
-contentRequest.send()
 
+4
+/*--------------- Table loading section ---------------*/
+let tableHeader = document.querySelector("thead")
+let tableBody = document.querySelector('tbody')
 
+function loadTable(headerObj) {
 
-// Read more functionality based on paragraph height
+  if (headerObj) {  // Checks whether a tale exist
+    let tableContainer = document.querySelector(".bottom-section")
+    const table = tableContainer.appendChild(document.createElement("table"))
+    table.appendChild(document.createElement("thead"))
+    table.appendChild(document.createElement("tbody"))
+
+    // Table header loading
+    tableFn.tableHeadLoader(headerObj)
+
+    // Table content loading
+    utils.jsonCaller("get", "resources/json/vacancies.json", function (object) {
+      contentObj = object
+      tableFn.tableBodyLoader(contentObj)
+    })
+
+  }
+}
 
+/*--------------- Paragraph resize and read more section ---------------*/
 const readMoreBtn = document.querySelector('.read-more')
 const readLessBtn = document.querySelector('.read-less')
 
-// Main resize function
-function paraResize(){
+
+function paraResize() {
   paraHeight = para.offsetHeight
-  if (paraHeight > 200) {
-    readLess();
+  if (paraHeight > 197) {
+    readLess()
   } else {
     para.style.height = 'auto'
     readMoreBtn.style.display = 'none'
     readLessBtn.style.display = 'none'
   }
 }
 
-// Read less
-function readLess(){
-  para.style.height = '200px'
+// Read less__
+readLessBtn.addEventListener('click', function () {
+  para.style.height = '197px'
+  readMoreBtn.style.display = 'block'
+  readLessBtn.style.display = 'none'
+})
+
+function readLess() {
+  para.style.height = '197px'
```

Comment:
Store value **197** in a **const** _heightThreshold_ in beginning of read-more resize section, to be referred to in below places. Avoid magic numbers in code.

Reply:
Removed magic numbers.